worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/shovel/ShovelDistributor.java

```diff
@@ -135,11 +155,10 @@ public void runOnce() {
 
                     final String shovelName = queueConsumptionTarget.getKey().getName();
 
-                    final boolean nonTerminatedShovelExists = retrievedShovels.stream()
-                            .anyMatch(s -> (s.getState() != ShovelState.TERMINATED) && (s.getName().endsWith(shovelName)));
+                    final boolean shovelExists = retrievedShovels.stream().anyMatch(s -> s.getName().endsWith(shovelName));
```

Comment:
This class used to handle trying to recreate shovels stuck in a terminated state, in addition to creating new shovels.

I have moved responsibility for recreating terminated shovels into the existing [NonRunningShovelChecker](https://github.com/WorkerFramework/worker-message-prioritization/pull/18/files#diff-44a22bec0da05c808153909de170d346fcb8dd9b1fc8c311953be9c3d416ad50R117) class, as I think it is a bit cleaner to have the shovel recovery code in one place, and means that we can restrict the use of the nodeSpecificShovelsApiCache to just the NonRunningShovelChecker and ShovelRunningTooLong classes.